src/Toolkit/Toolkit.Maui/BookmarksView/BookmarksView.cs

```diff
@@ -192,16 +170,36 @@ private void SelectAndNavigateToBookmark(Bookmark bookmark)
     /// <summary>
     /// Handles selection on the underlying list view.
     /// </summary>
-    private void Internal_bookmarkSelected(object? sender, SelectedItemChangedEventArgs e)
+    private void Internal_bookmarkSelected(object? sender, SelectionChangedEventArgs e)
+    {
+        if (sender is CollectionView cv)
+        {
+            if (cv.SelectedItem is Bookmark item)
+            {
+                SelectAndNavigateToBookmark(item);
+            }
+            cv.ClearValue(CollectionView.SelectedItemProperty);
+        }
+    }
+
+    private void UpdateListView()
     {
-        if (e.SelectedItem is Bookmark bm)
+        var collection = (CollectionView)GetTemplateChild(_presentingViewName);
+        if (collection != null)
         {
-            SelectAndNavigateToBookmark(bm);
+            collection.SelectionChanged -= Internal_bookmarkSelected;
+            collection.SelectionChanged += Internal_bookmarkSelected;
```

Comment:
This shouldn't be necessary - we should know when we need to subscribe and unsubscribe.